### PR TITLE
[Workflows] add a workflow that can be manually triggered on a PR.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Currently, we manually run the SLOW tests for PRs where applicable to ensure the changed components are robust enough to merge. This is burdensome. 

How about a workflow that we can trigger with a few clicks and run the desired tests within our CI runner?

This PR adds such a workflow.

### CODE REVIEW

File: .github/workflows/run_test_from_pr.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      pr_number: 
+        description: 'PR number'
+        required: true
+      docker_image:
+        default: 'diffusers/diffusers-pytorch-cuda'
+        description: 'Name of the Docker image'
+        required: true
+      test_command:
+        description: 'Test command to run (e.g.: `pytest tests/pipelines/dit/`). Any valid pytest command can be provided.'
+        required: true
```
Comment (BenjaminBossan) [inline .github/workflows/run_test_from_pr.yml:15]: Would it make sense to have a default here?

File: .github/workflows/run_test_from_pr.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      options: --gpus all --privileged --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
+
+    steps:
+      - name: NVIDIA-SMI
+        run: |
+          nvidia-smi
+
+      - name: Install `gh`
+        run: | 
+            (type -p wget >/dev/null || (apt update && apt-get install wget -y)) \
```
Comment (BenjaminBossan) [inline .github/workflows/run_test_from_pr.yml:41]: I guess this is based on: https://github.com/cli/cli/blob/trunk/docs/install_linux.md#official-sources, maybe could be added as a comment.

**Review by BenjaminBossan — APPROVED** [review]
Thanks for adding this convenient GH action, LGTM. I haven't worked with GH cli but it looks very useful for this case.

**Review by yiyixuxu — APPROVED** [review]
ahh this would be awesome!

File: .github/workflows/run_test_from_pr.yml (hunk 1 above)
Comment (sayakpaul) [reply .github/workflows/run_test_from_pr.yml:15]: I think not because having a default here wouldn't really help the case IMO.

**Pull request merged**